src/pal/src/thread/process.cpp

```diff
@@ -4313,43 +4310,47 @@ getPath(
     if (lpwstr)
     {
         /* convert path to multibyte, check buffer size */
-        n = WideCharToMultiByte(CP_ACP, 0, lpwstr, -1, lpPathFileName, iLen,
+        n = WideCharToMultiByte(CP_ACP, 0, lpwstr, -1, NULL, 0,
             NULL, NULL);
-        if (n == 0)
+        
+        if (!lpPathFileName.Reserve(n + lpFileNameString.GetCount() ))
         {
-            ASSERT("WideCharToMultiByte failure!\n");
+            ERROR("StackString Reserve failed!\n");
             return FALSE;
         }
 
-        n += strlen(lpFileName) + 2;
-        if (n > (INT)iLen)
-        {
-            ERROR("Buffer too small for full path!\n");
-            return FALSE;
-        }
+        lpPath = lpPathFileName.OpenStringBuffer(n);
+
+        n = WideCharToMultiByte(CP_ACP, 0, lpwstr, -1, lpPath, n,
+            NULL, NULL);
+        lpPathFileName.CloseBuffer(n - 1);
```

Comment:
In case the n is zero, you pass -1 to the CloseBuffer which would result in the m_count member being set to -1.